source/globalCommands.py

```diff
@@ -972,8 +929,21 @@ def script_review_activate(self,gesture):
 	script_review_activate.__doc__=_("Performs the default action on the current navigator object (example: presses it if it is a button).")
 	script_review_activate.category=SCRCAT_OBJECTNAVIGATION
 
+	def _moveWithBoundsChecking(self, info, unit, direction, endPoint=None):

```

Comment:
This could probably be moved into the base TextInfo class. 
Also, is there a reason why this is non-distructive (I.e. returns a copy)? I don't think you actually ever make use of this. It could just do it distructively and return res like the other move... unless you have a particular plan for this not yet done?

Reply:
This is a generalization of the console's old bounds checking code (i.e. we captured an `oldRange` and didn't update the underlying UIA text range if a move was impossible). Would destructive be okay here?